File: capacity-club-api/src/domain-modules/home-training/model/payload/home-training-update.payload.ts
import { UniqueId } from '@common/model';
import { Payload } from '@domain-modules-shared';
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsOptional, Length } from 'class-validator';
import { MemberHomeTraining } from 'domain-modules/member-home-training/model/entity/member_home_training.entity';
import { Workout } from 'domain-modules/workout/model/entity/workout.entity';

/**
 * Class used to update home training.
 * This class defines the payload used when updating existing home training.
 */
export class HomeTrainingUpdatePayload implements Payload {
  @ApiProperty()
  @IsNotEmpty()
  @Length(26, 26)
  home_training_id: UniqueId;
  @ApiProperty()
  @IsNotEmpty()
  title: string;
  @ApiProperty()
  @IsOptional()
  description: string;
  @ApiProperty()
  @IsNotEmpty()
  nb_week: number;
  @ApiProperty()
  @IsNotEmpty()
  nb_training_by_week: number;
  @ApiProperty()
  @IsNotEmpty()
  price: number;
  @ApiProperty({ type: () => Workout, isArray: true })
  @IsNotEmpty()
  workouts: Workout[];
  @ApiProperty({ type: () => MemberHomeTraining, isArray: true })
  @IsNotEmpty()
  member_home_trainings: MemberHomeTraining[];
}

File: capacity_club_mobile_app/lib/domain-features/data/home-training/model/payload/home_training_update_payload.dart
import 'package:capacity_club_mobile_app/core/model/abstract/payload.dart';
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';
import 'package:capacity_club_mobile_app/domain-features/data/workout/model/workout_model.dart';
import 'package:equatable/equatable.dart';

class HomeTrainingUpdatePayload extends Equatable implements Payload {
  final UniqueId home_training_id;
  final String title;
  final int nb_week;
  final int nb_training_by_week;
  final double price;
  final List<WorkoutModel> workouts;

  const HomeTrainingUpdatePayload({
    required this.home_training_id,
    required this.title,
    required this.nb_week,
    required this.nb_training_by_week,
    required this.price,
    required this.workouts,
  });

  Map<String, dynamic> toJson() => {
        'home_training_id': home_training_id.toJson(),
        'title': title,
        'nb_week': nb_week,
        'nb_training_by_week': nb_training_by_week,
        'price': price,
        'workouts': workouts.map((workout) => workout.toJson()).toList(),
      };

  @override
  List<Object?> get props => [
        home_training_id,
        title,
        nb_week,
        nb_training_by_week,
        price,
        workouts,
      ];
}
